packages/node-core/src/indexer/project.service.ts

```diff
@@ -283,7 +291,16 @@ export abstract class BaseProjectService<API extends IApi, DS extends BaseDataSo
         });
     }
 
-    return new BlockHeightMap(dsMap);
+    const newDsMap = cloneDeep(dsMap);
+    [...dsMap.entries()].forEach(([height, ds], i, arr) => {
+      const nextStart = arr[i + 1]?.[0];
+      const endBlock = maxEndBlockHeight(ds);
+      if (endBlock < Number.MAX_SAFE_INTEGER && (nextStart === undefined || nextStart > endBlock + 1)) {
+        newDsMap.set(endBlock + 1, []);
+      }
```

Comment:
This only seems to consider at the end. Can you please write tests for this.

I don't think this would work:

```ts
const dataSources =  [{ startBlock: 1}, { startBlock: 10, endBlock 20, }];

// before this pr
const map = new Map([
  [1, [/* ds 1*/]],
  [10, [/* ds 1 and 2 */],
])


// what it should be
const map = new Map([
  [1, [/* ds 1*/]],
  [10, [/* ds 1 and 2 */],
  [21, [/* ds 1*/]]
]);

// What it looks like its doing
const map = new Map([
  [1, [/* ds 1*/]],
  [10, [/* ds 1 and 2 */],
  [21, [/* empty */]]
]);
```

It also needs to accommodate more datasources that could start and end anywhere inbetween